src/apps/components/run.js

```diff
@@ -42,8 +43,13 @@ export class RunApp extends React.Component {
     componentWillReceiveProps(nextProps) {
         let self = this;
         let newBundle = nextProps.bundleId !== this.props.bundleId;
+        let isLocal = nextProps.bundleId === LOCAL_BUNDLE_ID;
         let bundleUpdated = newBundle || (nextProps.bundle !== this.props.bundle);
 
+        if (nextProps.error) {
+            this._viewStatusChange(ViewStatus.STOPPED);
+        }
```

Comment:
whoa, if you want to make sure the `viewStatus` goes to `STOPPED` you should call `view.stop` and let view fire a `viewStatus` change.